component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -1129,4 +1130,27 @@ public static Map<String, String> getClaimProperties(String tenantDomain, String
         }
         return Collections.emptyMap();
     }
+
+    /**
+     * Returns the encrypted claim value if EnableEncryption property is available for the claim. Otherwise, return the
+     * plain text.
+     *
+     * @param claimURI     Claim URI.
+     * @param claimValue   Claim value.
+     * @param tenantDomain Tenant domain.
+     * @return Encrypted or plain claim value based on the EnableEncryption claim property.
+     * @throws TOTPAuthenticatorException When errors occurs while encrypting the claim value.
+     */
+    public static String getProcessedClaimValue(String claimURI, String claimValue, String tenantDomain) {
+
+        Map<String, String> claimProperties = getClaimProperties(tenantDomain, claimURI);
+        try {
+            if (claimProperties.containsKey(IdentityMgtConstants.ENABLE_ENCRYPTION)) {
+                return claimValue;
+            }
+            return TOTPUtil.encrypt(claimValue);
```

Comment:
Is this null and empty safe ?

Reply:
Ideally, this cannot be null or empty, as this is a system-generated value. However, will add an empty check to be safe.

Reply:
Can we use StringUtil isNotBlank ?

Reply:
done